src/sql/workbench/contrib/executionPlan/browser/widgets/findExpensiveNodeWidget.ts

```diff
@@ -0,0 +1,204 @@
+			await new FindExpensiveOperationAction().run(self);
+		});
+
+		// Adds Action bar
+		this._actionBar = new ActionBar(this.container);
+		this._actionBar.context = this;
+		this._actionBar.pushAction(new CancelExpensiveOperationAction(), { label: false, icon: true });
+	}
+
+	public focus() {
+		this._operationNameSelectBox.focus();
+	}
+
+	public getExpensiveOperationDelegate(): (cell: AzDataGraphCell) => number | undefined {
+		const getElapsedTimeInMs = (cell: AzDataGraphCell): number | undefined => cell.elapsedTimeInMs;
+		const getElapsedCpuTimeInMs = (cell: AzDataGraphCell): number | undefined => cell.elapsedCpuTimeInMs;
+		const getCost = (cell: AzDataGraphCell): number | undefined => cell.cost;
+		const getSubtreeCost = (cell: AzDataGraphCell): number | undefined => cell.subTreeCost;
+
+		const getRowsForAllExecutions = (cell: AzDataGraphCell): number | undefined => {
+			if (!cell.rowMetrics) {
+				return undefined;
+			}
+
+			let result = Number(cell.rowMetrics['actualRows']);
+			if (!result) {
+				result = Number(cell.rowMetrics['estimateRowsAllExecs']);
+			}
+
+			if (isNaN(result)) {
+				return undefined;
+			}
+
+			return result;
+		};
+
+		const getNumberOfRowsRead = (cell: AzDataGraphCell): number | undefined => {
+			if (!cell.rowMetrics) {
+				return undefined;
+			}
+
+			let result = Number(cell.rowMetrics['actualRowsRead']);
+			if (!result) {
+				result = Number(cell.rowMetrics['estimatedRowsRead']);
+			}
+
+			if (isNaN(result)) {
+				return undefined;
+			}
+
+			return result;
+		};
+
+		switch (this._selectedExpensiveOperationType) {
+			case ExpensiveOperationType.ActualElapsedTime:
+				return getElapsedTimeInMs;
+			case ExpensiveOperationType.ActualElapsedCpuTime:
+				return getElapsedCpuTimeInMs;
+			case ExpensiveOperationType.Cost:
+				return getCost;
+			case ExpensiveOperationType.SubtreeCost:
+				return getSubtreeCost;
+			case ExpensiveOperationType.ActualNumberOfRowsForAllExecutions:
+				return getRowsForAllExecutions;
+			case ExpensiveOperationType.NumberOfRowsRead:
+				return getNumberOfRowsRead;
+		}
+	}
+}
+
+export class FindExpensiveOperationAction extends Action {
+	public static ID = 'qp.findExpensiveOperationAction';
+	public static LABEL = localize('findExpensiveOperationAction', 'Find (Enter)');
+
+	constructor() {
+		super(FindExpensiveOperationAction.ID, FindExpensiveOperationAction.LABEL, searchIconClassNames);
+	}
+
+	public override async run(context: FindExpensiveOperationWidget): Promise<void> {
+		const expensiveOperationDelegate: (cell: AzDataGraphCell) => number | undefined = context.getExpensiveOperationDelegate();
+
+		context.executionPlanDiagram.clearExpensiveOperatorHighlighting();
+		let result = context.executionPlanDiagram.highlightExpensiveOperator(expensiveOperationDelegate);
+		if (!result) {
+			context.notificationService.info(localize('invalidPropertyExecutionPlanMetric', 'Unable to locate a node using the specified metric.'));
```

Comment:
I would suggest saying what the metric was too - and maybe reword it to be something like `No nodes found with the '${MetricName}' metric.`

I'm also not a huge fan of the toast notifications being used for this. Taking the find widget as inspiration it would be better to have the widget itself indicate that nothing was found.

Reply:
this should be a warning or error instead of info, right?